### PR TITLE
Refactor admin employee part

### PR DESCRIPTION
**Pull request by FrankMerema**
- [x] fix add employee
- [x] Fix bug when projects being selected and other changes are done, the projects will be resetted

### CODE REVIEW

File: pages/admin/employees/add.vue [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+</template>
+
+<script lang="ts">
+import {defineComponent, ref} from "@nuxtjs/composition-api";
+
+export default defineComponent({
+  middleware: ["isAdmin"],
+  setup() {
+    const employee = ref({});
+    const employeeFormMode = 'add';
```
Comment (sanderdejong88) [inline pages/admin/employees/add.vue:12]: Not used in template

File: pages/admin/employees/add.vue [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  <employee-form mode="add" :employee="employee" />
+</template>
+
+<script lang="ts">
+import {defineComponent, ref} from "@nuxtjs/composition-api";
+
+export default defineComponent({
+  middleware: ["isAdmin"],
+  setup() {
+    const employee = ref({});
```
Comment (sanderdejong88) [inline pages/admin/employees/add.vue:11]: Why pass in an empty object?

Comment (FrankMerema) [reply pages/admin/employees/add.vue:11]: If you want to create a new employee you need to have an object as a base to start filling in the props :) That's how the app is built now 😉

File: components/employees/employee-form.vue [hunk 3]
```diff
@@ -201,228 +90,77 @@ import {
...
     );
 
-    const isAdmin = ref<boolean>(
-      store.getters["employees/adminList"].includes(employee.value?.email)
-    );
+    // We probably shouldn't watch a store getter imho
     watch(
       () =>
-        store.getters["employees/adminList"].includes(employee.value?.email),
+        store.getters["employees/adminList"].includes(props.employee?.email),
```
Comment (sanderdejong88) [inline components/employees/employee-form.vue:151]: duplicate line + why the includes part in the watch? shouldnt it just watch the list?

Comment (FrankMerema) [reply components/employees/employee-form.vue:151]: I have no idea, is existing code, and didn't touch it, except for the object to look at props

**Review by sanderdejong88 — APPROVED** [review]

**Pull request merged**